[ORTModule] unexpected tensor scalar type in onnx export

**Describe the bug**

I tried `ORTModule` for the model below (I have included the test code for completeness) but I get the error:
> RuntimeError: There was an error while exporting the PyTorch model to ONNX: unexpected tensor scalar type

Please note that if I directly call `torch.onnx.export`, I don't get this error (see the console output below). This makes me think it is an ORTModule issue.


```python
# test_ort.py
import os
from typing import Any, Callable

import torch
import torch.nn as nn
import torch.nn.functional as F
import torch.onnx
from torch.onnx import TrainingMode
from torch.onnx import OperatorExportTypes

from torch_ort import ORTModule


class ConvNet(nn.Module):
    def __init__(self, use_dropout=True):
        super().__init__()
        self.conv1 = nn.Conv2d(1, 32, 3, 1)
        self.conv2 = nn.Conv2d(32, 64, 3, 1)
        self.dropout1 = nn.Dropout(0.25 if use_dropout else 0.0)
        self.dropout2 = nn.Dropout(0.5 if use_dropout else 0.0)
        self.fc1 = nn.Linear(9216, 128)
        self.fc2 = nn.Linear(128, 10)

    def forward(self, x):
        x = self.conv1(x)
        x = F.relu(x)
        x = self.conv2(x)
        x = F.relu(x)
        x = F.max_pool2d(x, 2)
        x = self.dropout1(x)
        x = torch.flatten(x, 1)
        x = self.fc1(x)
        x = F.relu(x)
        x = self.dropout2(x)
        x = self.fc2(x)
        output = F.log_softmax(x, dim=1)
        return output


def _get_model_and_inptus(cls: Callable[[Any], torch.nn.Module], batch_size, d_in, d_out, **kwargs):
    model = cls(**kwargs)
    if isinstance(d_in, int):
        d_in = (d_in,)
    inputs = (torch.randn(batch_size, *d_in),)
    labels = torch.randint(0, d_out - 1, (batch_size,))
    return model, inputs, labels


def get_model_and_inputs():
    return _get_model_and_inptus(ConvNet, 64, (1, 28, 28), 10)


def test_ortmodule_from_pytorch_conv_net():
    model, sample_inputs, _ = get_model_and_inputs()
    torch.onnx.export(
        model,
        sample_inputs,
        "exported.onnx",
        training=TrainingMode.TRAINING,
        opset_version=12,
        operator_export_type=OperatorExportTypes.ONNX,
        verbose=True,
    )
    os.environ["ORTMODULE_FALLBACK_POLICY"] = "FALLBACK_DISABLE"
    model = ORTModule(model)
    _ = model(*sample_inputs)
```

**Urgency**
none

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 20.04.3 LTS
- ONNX Runtime installed from (source or binary): onnxruntime-training==1.9.0+cu111
- ONNX Runtime version:
- Python version: Python 3.8.10
- Visual Studio version (if applicable):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: 11.1.1
- GPU model and memory: V100 32 GB

**To Reproduce**

Dockerfile:
```
FROM nvcr.io/nvidia/cuda:11.1.1-devel-ubuntu20.04
RUN apt-get update
RUN apt-get install -y --no-install-recommends \
    python3 python3-pip python3-dev
RUN update-alternatives --install /usr/bin/python python /usr/bin/python3 1
RUN update-alternatives --install /usr/bin/pip pip /usr/bin/pip3 1
RUN pip install --upgrade pip
RUN python --version
RUN pip --version
RUN pip install onnxruntime-training==1.9.0+cu111 -f https://download.onnxruntime.ai/onnxruntime_stable_cu111.html
RUN pip install torch-ort==1.9.0 ninja

RUN python -m torch_ort.configure

RUN pip install pytest

WORKDIR /workspace

COPY *.py /workspace/
```

Steps to reproduce:
```
pytest test_ort.py # (the file above)
```


**Expected behavior**
No error in converting pytorch to onnx and running it through ORTModule.

**Screenshots**

```
root@cc4d845f904d:/workspace# pytest test_ort.py -sv
============================================================================================== test session starts ==============================================================================================
platform linux -- Python 3.8.10, pytest-6.2.5, py-1.10.0, pluggy-1.0.0 -- /usr/bin/python3
cachedir: .pytest_cache
rootdir: /workspace
collected 1 item

test_ort.py::test_ortmodule_from_pytorch_conv_net graph(%input.1 : Float(64, 1, 28, 28, strides=[784, 784, 28, 1], requires_grad=0, device=cpu),
      %conv1.weight : Float(32, 1, 3, 3, strides=[9, 9, 3, 1], requires_grad=1, device=cpu),
      %conv1.bias : Float(32, strides=[1], requires_grad=1, device=cpu),
      %conv2.weight : Float(64, 32, 3, 3, strides=[288, 9, 3, 1], requires_grad=1, device=cpu),
      %conv2.bias : Float(64, strides=[1], requires_grad=1, device=cpu),
      %fc1.weight : Float(128, 9216, strides=[9216, 1], requires_grad=1, device=cpu),
      %fc1.bias : Float(128, strides=[1], requires_grad=1, device=cpu),
      %fc2.weight : Float(10, 128, strides=[128, 1], requires_grad=1, device=cpu),
      %fc2.bias : Float(10, strides=[1], requires_grad=1, device=cpu)):
  %9 : Float(64, 32, 26, 26, strides=[21632, 676, 26, 1], requires_grad=1, device=cpu) = onnx::Conv[dilations=[1, 1], group=1, kernel_shape=[3, 3], pads=[0, 0, 0, 0], strides=[1, 1]](%input.1, %conv1.weight, %conv1.bias) # /usr/local/lib/python3.8/dist-packages/torch/nn/modules/conv.py:439:0
  %10 : Float(64, 32, 26, 26, strides=[21632, 676, 26, 1], requires_grad=1, device=cpu) = onnx::Relu(%9) # /usr/local/lib/python3.8/dist-packages/torch/nn/functional.py:1298:0
  %11 : Float(64, 64, 24, 24, strides=[36864, 576, 24, 1], requires_grad=1, device=cpu) = onnx::Conv[dilations=[1, 1], group=1, kernel_shape=[3, 3], pads=[0, 0, 0, 0], strides=[1, 1]](%10, %conv2.weight, %conv2.bias) # /usr/local/lib/python3.8/dist-packages/torch/nn/modules/conv.py:439:0
  %12 : Float(64, 64, 24, 24, strides=[36864, 576, 24, 1], requires_grad=1, device=cpu) = onnx::Relu(%11) # /usr/local/lib/python3.8/dist-packages/torch/nn/functional.py:1298:0
  %13 : Float(64, 64, 12, 12, strides=[9216, 144, 12, 1], requires_grad=1, device=cpu) = onnx::MaxPool[ceil_mode=0, kernel_shape=[2, 2], pads=[0, 0, 0, 0], strides=[2, 2]](%12) # /usr/local/lib/python3.8/dist-packages/torch/nn/functional.py:718:0
  %14 : Float(device=cpu) = onnx::Constant[value={0.25}]()
  %15 : Bool(device=cpu) = onnx::Constant[value={1}]()
  %16 : Float(64, 64, 12, 12, strides=[9216, 144, 12, 1], requires_grad=1, device=cpu), %17 : Bool(64, 64, 12, 12, strides=[9216, 144, 12, 1], device=cpu) = onnx::Dropout(%13, %14, %15) # /usr/local/lib/python3.8/dist-packages/torch/nn/functional.py:1168:0
  %18 : Float(64, 9216, strides=[9216, 1], requires_grad=1, device=cpu) = onnx::Flatten[axis=1](%16) # /workspace/test_ort.py:31:0
  %19 : Float(64, 128, strides=[128, 1], requires_grad=1, device=cpu) = onnx::Gemm[alpha=1., beta=1., transB=1](%18, %fc1.weight, %fc1.bias) # /usr/local/lib/python3.8/dist-packages/torch/nn/functional.py:1847:0
  %20 : Float(64, 128, strides=[128, 1], requires_grad=1, device=cpu) = onnx::Relu(%19) # /usr/local/lib/python3.8/dist-packages/torch/nn/functional.py:1298:0
  %21 : Float(device=cpu) = onnx::Constant[value={0.5}]()
  %22 : Bool(device=cpu) = onnx::Constant[value={1}]()
  %23 : Float(64, 128, strides=[128, 1], requires_grad=1, device=cpu), %24 : Bool(64, 128, strides=[128, 1], device=cpu) = onnx::Dropout(%20, %21, %22) # /usr/local/lib/python3.8/dist-packages/torch/nn/functional.py:1168:0
  %25 : Float(64, 10, strides=[10, 1], requires_grad=1, device=cpu) = onnx::Gemm[alpha=1., beta=1., transB=1](%23, %fc2.weight, %fc2.bias) # /usr/local/lib/python3.8/dist-packages/torch/nn/functional.py:1847:0
  %26 : Float(64, 10, strides=[10, 1], requires_grad=1, device=cpu) = onnx::LogSoftmax[axis=1](%25) # /usr/local/lib/python3.8/dist-packages/torch/nn/functional.py:1768:0
  return (%26)

FAILED

=================================================================================================== FAILURES ====================================================================================================
_____________________________________________________________________________________ test_ortmodule_from_pytorch_conv_net ______________________________________________________________________________________
RuntimeError: There was an error while exporting the PyTorch model to ONNX: unexpected tensor scalar type

The above exception was the direct cause of the following exception:

    def test_ortmodule_from_pytorch_conv_net():
        model, sample_inputs, _ = get_model_and_inputs()
        # sample_inputs[0].requires_grad = True
        torch.onnx.export(
            model,
            sample_inputs,
            "exported.onnx",
            training=TrainingMode.TRAINING,
            opset_version=12,
            operator_export_type=OperatorExportTypes.ONNX,
            verbose=True,
        )
        os.environ["ORTMODULE_FALLBACK_POLICY"] = "FALLBACK_DISABLE"
        model = ORTModule(model)
>       _ = model(*sample_inputs)

test_ort.py:67:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/usr/local/lib/python3.8/dist-packages/torch/nn/modules/module.py:1051: in _call_impl
    return forward_call(*input, **kwargs)
/usr/local/lib/python3.8/dist-packages/onnxruntime/training/ortmodule/ortmodule.py:81: in _forward
    return self._torch_module.forward(*inputs, **kwargs)
/usr/local/lib/python3.8/dist-packages/onnxruntime/training/ortmodule/_torch_module_ort.py:32: in _forward
    return self._execution_manager(self.is_training()).forward(*inputs, **kwargs)
/usr/local/lib/python3.8/dist-packages/onnxruntime/training/ortmodule/_training_manager.py:259: in forward
    self._fallback_manager.handle_exception(exception=e,
/usr/local/lib/python3.8/dist-packages/onnxruntime/training/ortmodule/_fallback.py:194: in handle_exception
    raise exception
/usr/local/lib/python3.8/dist-packages/onnxruntime/training/ortmodule/_training_manager.py:82: in forward
    build_gradient_graph = self._export_model(*inputs, **kwargs)
/usr/local/lib/python3.8/dist-packages/onnxruntime/training/ortmodule/_graph_execution_manager.py:294: in _export_model
    self._onnx_models.exported_model = self._get_exported_model(
/usr/local/lib/python3.8/dist-packages/onnxruntime/training/ortmodule/_graph_execution_manager.py:361: in _get_exported_model
    raise wrap_exception(ORTModuleONNXModelException,
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

new_exception = <class 'onnxruntime.training.ortmodule._fallback.ORTModuleONNXModelException'>
raised_exception = RuntimeError('There was an error while exporting the PyTorch model to ONNX: unexpected tensor scalar type')

    def wrap_exception(new_exception: ORTModuleFallbackException, raised_exception: Exception) -> ORTModuleFallbackException:
        '''Wraps `raised_exception` exception as cause for the returned `new_exception` exception'''

        exception = None
        try:
>           raise new_exception(raised_exception) from raised_exception
E           onnxruntime.training.ortmodule._fallback.ORTModuleONNXModelException: There was an error while exporting the PyTorch model to ONNX: unexpected tensor scalar type

/usr/local/lib/python3.8/dist-packages/onnxruntime/training/ortmodule/_fallback.py:226: ORTModuleONNXModelException
=============================================================================================== warnings summary ================================================================================================
../usr/local/lib/python3.8/dist-packages/onnx/mapping.py:27
  /usr/local/lib/python3.8/dist-packages/onnx/mapping.py:27: DeprecationWarning: `np.object` is a deprecated alias for the builtin `object`. To silence this warning, use `object` by itself. Doing this will not modify any behavior and is safe.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    int(TensorProto.STRING): np.dtype(np.object)

test_ort.py::test_ortmodule_from_pytorch_conv_net
  /usr/local/lib/python3.8/dist-packages/torch/onnx/utils.py:298: UserWarning: It is recommended that constant folding be turned off ('do_constant_folding=False') when exporting the model in training-amenable mode, i.e. with 'training=TrainingMode.TRAIN' or 'training=TrainingMode.PRESERVE' (when model is in training mode). Otherwise, some learnable model parameters may not translate correctly in the exported ONNX model because constant folding mutates model parameters. Please consider turning off constant folding or setting the training=TrainingMode.EVAL.
    warnings.warn("It is recommended that constant folding be turned off ('do_constant_folding=False') "

-- Docs: https://docs.pytest.org/en/stable/warnings.html
============================================================================================ short test summary info ============================================================================================
FAILED test_ort.py::test_ortmodule_from_pytorch_conv_net - onnxruntime.training.ortmodule._fallback.ORTModuleONNXModelException: There was an error while exporting the PyTorch model to ONNX: unexpected tens...
========================================================================================= 1 failed, 2 warnings in 2.44s =========================================================================================
```

**Additional context**
pip freeze
> attrs==21.2.0
Cerberus==1.3.4
flatbuffers==2.0
h5py==3.4.0
iniconfig==1.1.1
mpmath==1.2.1
ninja==1.10.2
numpy==1.21.2
onnx==1.10.1
onnxruntime-training==1.9.0+cu111
packaging==21.0
pluggy==1.0.0
protobuf==3.18.0
py==1.10.0
pyparsing==2.4.7
pytest==6.2.5
six==1.16.0
sympy==1.8
toml==0.10.2
torch==1.9.1
torch-ort==1.9.0
typing-extensions==3.10.0.2

--- a/orttraining/orttraining/python/training/ortmodule/_custom_op_symbolic_registry.py
+++ b/orttraining/orttraining/python/training/ortmodule/_custom_op_symbolic_registry.py
@@ -87,6 +87,9 @@ def multinomial(g, self, num_samples, replacement=False, generator=None):
 
 @register_symbolic('max_pool2d')
 def max_pool2d(g, self, kernel_size, stride, padding, dilation, ceil_mode):
+    stride_val = sym_help._maybe_get_const(stride, 'is')
+    if not stride_val:
+        stride = kernel_size
     return g.op("com.microsoft::ATenOp", self, kernel_size, stride, padding, dilation, ceil_mode,
                 name_s='aten::max_pool2d_with_indices', outputs=2)[0]
 
@@ -103,6 +106,9 @@ def argmax(g, input, dim, keepdim):
 
 @register_symbolic('avg_pool2d')
 def avg_pool2d(g, self, kernel_size, stride, padding, ceil_mode, count_include_pad, divisor_override):
+    stride_val = sym_help._maybe_get_const(stride, 'is')
+    if not stride_val:
+        stride = kernel_size
     return g.op("com.microsoft::ATenOp", self, kernel_size, stride, padding, ceil_mode,
                 count_include_pad, divisor_override, name_s='aten::avg_pool2d')
 

--- a/orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py
+++ b/orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py
@@ -916,6 +916,40 @@ def test_gradient_correctness_pool2d(pool_type):
         _test_helpers.assert_values_are_close(ort_prediction, pt_prediction)
         _test_helpers.assert_gradients_match_and_reset_gradient(ort_model, pt_model, rtol=5e-3, atol=4e-3)
 
+@pytest.mark.parametrize("pool_type", ['MaxPool', 'AvgPool'])
+@pytest.mark.parametrize("stride", [None, 2])
+def test_export_correctness_pool2d(pool_type, stride):
+    class NeuralNetPool2d(torch.nn.Module):
+        def __init__(self):
+            super(NeuralNetPool2d, self).__init__()
+            self.conv = torch.nn.Conv2d(3, 64, kernel_size=7, stride=2, padding=3, bias=False)
+            self.pool_type = pool_type
+            
+
+        def forward(self, input):
+            x = self.conv(input)
+            if pool_type == 'MaxPool':
+                output = torch.nn.functional.max_pool2d(x, kernel_size=3, stride=stride)
+            elif pool_type == 'AvgPool':
+                output = torch.nn.functional.avg_pool2d(x, kernel_size=3, stride=stride)
+            return output
+
+    N, C, H, W = 8, 3, 224, 224
+    device = 'cuda'
+    pt_model = NeuralNetPool2d().to(device)
+    ort_model = ORTModule(copy.deepcopy(pt_model))
+
+    def run_step(model, input):
+        prediction = model(input)
+        return prediction
+
+    for _ in range(10):
+        input = torch.randn(N, C, H, W, device=device)
+        pt_prediction = run_step(pt_model, input)
+        ort_prediction = run_step(ort_model, input)
+
+        _test_helpers.assert_values_are_close(ort_prediction, pt_prediction)
+
 def test_gradient_correctness_argmax_unfold():
     class NeuralNetUnfold(torch.nn.Module):
         def __init__(self, input_size, hidden_size, unfold_dim, unfold_size, unfold_step):

